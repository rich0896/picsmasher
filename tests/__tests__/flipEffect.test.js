const { FlipEffect } = require('../../src/effects.js');
const { createTestCanvas } = require('./helpers/testHelpers.js');

describe('FlipEffect', () => {
    test('should flip the image horizontally', async () => {
        const { canvas, ctx } = await createTestCanvas();
        const effect = new FlipEffect({ direction: 'horizontal' });

        const consoleSpy = jest
            .spyOn(console, 'log')
            .mockImplementation(() => {});

        effect.apply(ctx, canvas);

        expect(consoleSpy).toHaveBeenCalledWith('Applying Flip Effect');
        consoleSpy.mockRestore();

        const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
        expect(imageData).toMatchSnapshot();
    });

    test('should flip the image vertically', async () => {
        const { canvas, ctx } = await createTestCanvas();
        const effect = new FlipEffect({ direction: 'vertical' });

        const consoleSpy = jest
            .spyOn(console, 'log')
            .mockImplementation(() => {});

        effect.apply(ctx, canvas);

        expect(consoleSpy).toHaveBeenCalledWith('Applying Flip Effect');
        consoleSpy.mockRestore();

        const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
        expect(imageData).toMatchSnapshot();
    });
});
