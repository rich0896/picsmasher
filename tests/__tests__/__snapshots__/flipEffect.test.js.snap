// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlipEffect should flip the image horizontally 1`] = `ImageData { width: 200, height: 200, data: [100, 150, 200, 255, 100, 150, 200, 255, 100, 150...], length: 160000 }`;

exports[`FlipEffect should flip the image vertically 1`] = `ImageData { width: 200, height: 200, data: [100, 150, 200, 255, 100, 150, 200, 255, 100, 150...], length: 160000 }`;
